iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/OperatorTreeGenerator.java

```diff
@@ -2535,7 +2588,7 @@ private List<TSDataType> getOutputColumnTypesOfTimeJoinNode(PlanNode node) {
         dataTypes.add(((SeriesScanNode) child).getSeriesPath().getSeriesType());
       } else if (child instanceof AlignedSeriesScanNode) {
         dataTypes.add(((AlignedSeriesScanNode) child).getAlignedPath().getSeriesType());
-      } else if (child instanceof TimeJoinNode) {
+      } else if (child instanceof FullOuterTimeJoinNode) {
```

Comment:
Maybe we need change to `child instanceof FullOuterTimeJoinNode or child instanceof LeftOuterTimeJoinNode or child instanceof InnerTimeJoinNode`. 
I will test template + align by device query when the function is completed later.